src/components/features/AddPostForm.js

```diff
@@ -0,0 +1,89 @@
+import clsx from 'clsx';
+import styles from './AddPostForm.module.scss';
+import { Form, Button } from 'react-bootstrap';
+import { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { addPost } from '../../Redux/postsRedux';
+import { useNavigate } from 'react-router-dom';
+
+const AddPostForm = () => {
+
+  const [title, setTitle] = useState('');
+  const [author, setAuthor] = useState('');
+  const [publishedDate, setPublishedDate] = useState('');
+  const [shortDescription, setShortDescription] = useState('');
+  const [content, setContent] = useState('');
+
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+
+  const handleSubmit = e => {
+    e.preventDefault();
+    dispatch(addPost({ title, author, publishedDate, shortDescription, content }))
+    setTitle('');
+    setAuthor('');
+    setPublishedDate('');
+    setShortDescription('');
+    setContent('');
+    navigate("/", { replace: true });
+  }
+
+  return (
+    <Form onSubmit={handleSubmit} className={styles.form}>
+      <Form.Group className={clsx("mb-3", styles.smallInput)} controlId="formTitle">
```

Comment:
Tak widzę, że ta struktura pojedynczej kontrolki jest bardzo podobna dla wszystkich przypadków użycia. Może warto rozważyć, aby stworzyć osobny komponent do którego za pomocą propsów przekażesz to co się tam zmienia?

Reply:
Utworzyłem komponent PostForm.js , przerobiłem i przeniosłem kod z logiką dodawania nowego postu z AddPostForm.js do nowego komponentu PostForm.js. Mam nadzieję, że o to chodziło :)

Reply:
Nie do końca to miałem na myśli. Chodziło mi o to, że pojedyczna kontrolka formularza ma podobną strukturę:
```     <Group><Label></Label><Control/></Group>```, więc warto rozważyć stworzenie np. osobnego komponentu, np. FormElement, gdzie po prostu przekazywałbyś rzeczy które się różnią. Wtedy miałbyś po prostu kompozycję złożoną z kilku FormElement w AddPostForm